src/plugins/str.rs

```diff
@@ -33,6 +35,26 @@ impl Str {
         let applied = match self.action.as_ref() {
             Some(Action::Downcase) => Value::string(input.to_ascii_lowercase()),
             Some(Action::Upcase) => Value::string(input.to_ascii_uppercase()),
+            Some(Action::Substring(s)) => {
+                // Index operator isn't perfect: https://users.rust-lang.org/t/how-to-get-a-substring-of-a-string/1351
+                let no_spaces: String = s.chars().filter(|c| !c.is_whitespace()).collect();
+                let v: Vec<&str> = no_spaces.split(',').collect();
+                let start: usize = match v[0] {
+                    "" => 0,
+                    _ => v[0].parse().unwrap(),
+                };
+                let end: usize = match v[1] {
+                    "" => input.len(),
+                    _ => cmp::min(v[1].parse().unwrap(), input.len()),
+                };
+                if start > input.len() - 1 {
+                    Value::string("")
+                } else if start > end {
+                    Value::string(input)
```

Comment:
I wonder if we should return an error in the case where start comes after end.

Reply:
For the life of me I could not figure out how to test an error coming out of `Plugin::filter`. Every de-structuring/unwrapping/crying/pleading I tried failed... Then I realized that's not where it belongs anyway and now we have a much more logical approach to sub-parameters!